src/endpoint-with-defaults.ts

```diff
@@ -3,6 +3,14 @@ export = endpointWithDefaults
 import merge = require('./merge')
 import parse = require('./parse')
 
-function endpointWithDefaults (defaults: typeof import("./defaults"), route: string, options: { method?: string; url?: string; [key: string]: any }) {
-  return parse(merge(defaults, route!, options))
+import { EndpointDefaultOptions, EndpointOptions, RequestOptions, Route, RouteOptions } from './types'
+
+function endpointWithDefaults (defaults: EndpointDefaultOptions, route: Route, options?: RouteOptions): RequestOptions
+function endpointWithDefaults (defaults: EndpointDefaultOptions, options: EndpointOptions): RequestOptions
```

Comment:
Ah, I think I figured it out a minute after I added my question (because of course). We can remove the overloads above, because they are already defined in `types.ts#endpoint`